src/transformers/generation/utils.py

```diff
@@ -781,6 +781,8 @@ def _prepare_decoder_input_ids_for_generation(
         # exception: Donut checkpoints have task-specific decoder starts and don't expect a BOS token
         elif self.config.model_type == "vision-encoder-decoder" and "donut" in self.name_or_path.lower():
             pass
+        elif self.config.model_type in ["whisper"]:
```

Comment:
For Whisper we often don't start with the `decoder_input_ids` (e.g. when conditioning on the previous prompt OR when using "prompt_ids"). We should correct this here.